src/core/derivation.ts

```diff
@@ -181,9 +197,27 @@ export function trackDerivedFunction<T>(derivation: IDerivation, f: () => T, con
     }
     globalState.trackingDerivation = prevTracking
     bindDependencies(derivation)
+
+    if (derivation.observing.length === 0) {
```

Comment:
would be nice to move the production check to here for better code elimination (if the warning function is called only once)

Reply:
Done!

Reply:
Done